src/modules/cpu/kernel/normalize_generic.hpp

```diff
@@ -0,0 +1,513 @@
+            __m256 xy2 = _mm256_mul_ps(x, y2);
+            __m256 three_minus_xy2 = _mm256_sub_ps(avx_p3, xy2);
+            y = _mm256_mul_ps(y, three_minus_xy2);
+            y = _mm256_mul_ps(y, mulx8);
+            _mm256_storeu_ps(&input[i], y);
+        }
+    }
+    else
+    {
+        for (; i + 8 <= numElements; i += 8)
+        {
+            __m256 x = _mm256_loadu_ps(&input[i]);
+            x = _mm256_mul_ps(x, rdivx8);
+            __m256 mask = _mm256_cmp_ps(x, avx_p0, _CMP_NEQ_OQ);
+            __m256 y = _mm256_rsqrt_ps(x);
+            y = _mm256_and_ps(y, mask);
+            __m256 y2 = _mm256_mul_ps(y, y);
+            __m256 xy2 = _mm256_mul_ps(x, y2);
+            __m256 three_minus_xy2 = _mm256_sub_ps(avx_p3, xy2);
+            y = _mm256_mul_ps(y, three_minus_xy2);
+            y = _mm256_mul_ps(y, mulx8);
+            _mm256_storeu_ps(&input[i], y);
+        }
+    }
+    if (eps)
+    {
+        for (; i < numElements; i++)
+            input[i] = rpp_rsqrt(input[i] * rdiv + eps) * mul;
+    }
+    else
+    {
+        for (; i < numElements; i++)
+        {
+            Rpp32f x = input[i] * rdiv;
+            input[i] = x ? rpp_rsqrt(x) * mul : 0;
+        }
+    }
+}
+
+void compute_2D_mean(Rpp32f *srcPtr, Rpp32f *meanPtr, Rpp32u *dims, Rpp32u *stride)
+{
+    Rpp32f *srcPtrTemp = srcPtr;
+    Rpp32f normFactor = 1.0 / dims[1];
+    for(Rpp32u i = 0; i < dims[0]; i++)
+    {
+        meanPtr[i] = 0;
+        compute_sum(meanPtr[i], srcPtrTemp, 1, dims[1]);
+        srcPtrTemp += stride[1];
+        meanPtr[i] *= normFactor;
+    }
+}
+
+void compute_2D_inv_std_dev(Rpp32f *srcPtr, Rpp32f *meanPtr, Rpp32f *stdDevPtr, Rpp32u *dims, Rpp32u *stride) {
+
+    Rpp32f *srcPtrTemp = srcPtr;
+    Rpp32f normFactor = (Rpp32f)(1.0 / dims[1]);
+    for(Rpp32u i = 0; i < dims[0]; i++)
+    {
+        stdDevPtr[i] = 0;
+        compute_diff_square_sum(stdDevPtr[i], srcPtrTemp, 1, dims[1], meanPtr[i]);
+        srcPtrTemp += stride[1];
+    }
+    rpp_rsqrt_avx(stdDevPtr, (Rpp32s)dims[0], 0, normFactor, 1);
+}
+
+void normalize_3D_tensor_axis1_nontoggle(Rpp32f *srcPtr, RpptGenericDescPtr srcGenericDescPtr, Rpp32f *dstPtr, RpptGenericDescPtr dstGenericDescPtr,
+                         Rpp32f *meanPtr, Rpp32f *stdDevPtr, Rpp32f scale, Rpp32f shift, Rpp32u *paramStride)
+{
+    Rpp32f *srcPtrTemp = srcPtr;
+    Rpp32f *dstPtrTemp = dstPtr;
+    Rpp32s paramIdx = 0;
+    Rpp32f multiplier[srcGenericDescPtr->dims[3]];
+
+    for(int i = 0; i < srcGenericDescPtr->dims[3]; i++)
+        multiplier[i] = scale / stdDevPtr[i];
+
+    for(Rpp32u i = 0; i < srcGenericDescPtr->dims[1]; i++)
+    {
+        Rpp32f *srcPtrRow = srcPtrTemp;
+        Rpp32f *dstPtrRow = dstPtrTemp;
+        for(Rpp32u j = 0; j < srcGenericDescPtr->dims[2]; j++)
+        {
+            Rpp32f *srcPtrRowTemp = srcPtrRow;
+            Rpp32f *dstPtrRowTemp = dstPtrRow;
+            for(Rpp32u k = 0; k < srcGenericDescPtr->dims[3]; k++)
```

Comment:
Please try to vectorize the innermost loop
Lets say for if the last dimension is 1000, there will be 1000 mean and 1000 std values
This can be vectorized. Please check on this

Reply:
Sure, I have done this for our training case, but this is raw c code.